include/mum_finder.hpp

```diff
@@ -309,6 +318,25 @@ class mum_finder{
         return mum_subset;
     }
 
+    // write the mum to file
+    inline size_t write_mum(size_t const j, std::vector<std::pair<int, int>> const &idxs)
+    {
+        size_t count = 0;
+        int idx;
+        size_t mum_length;
+        // TODO
+        // j -= num_docs;
+        for (auto data : idxs){
```

Comment:
If I cared, you can pack these into a char* buffer to limit to one IO. Likely what you want.

Reply:
I've done something like this before; here's other code. 

`
        
        char *buffer = new char[total_size];

        size_t cursor = 0;
        size_t temp = character;
        memcpy(buffer + cursor, &temp, sizeof(uchar));
        cursor += sizeof(uchar);

        temp = idx;
        memcpy(buffer + cursor, &temp, BWT_BYTES);
        cursor += BWT_BYTES;

        temp = interval;
        memcpy(buffer + cursor, &temp, BWT_BYTES);
        cursor += BWT_BYTES;

        temp = offset;
        memcpy(buffer + cursor, &temp, BWT_BYTES);
        cursor += BWT_BYTES;

        temp = col_id;
        memcpy(buffer + cursor, &temp, ID_BYTES);
        cursor += ID_BYTES;

        temp = threshold;
        memcpy(buffer + cursor, &temp, BWT_BYTES);

        out.write(buffer, total_size);
        
        delete[] buffer;`